core/src/main/java/tc/oc/pgm/command/util/CommandGraph.java

```diff
@@ -85,9 +86,11 @@ protected PaperCommandManager<CommandSender> createCommandManager() throws Excep
         context ->
             context.getCommandContext().store(CommandKeys.INPUT_QUEUE, context.getInputQueue()));
 
-    // By default, suggestions run by a filtered processor.
-    // By default, it prevents suggestions like "s" -> "Something" or "someh" -> "Something"
-    manager.commandSuggestionProcessor((cpc, strings) -> strings);
+    // Basic suggestion filtering processor which avoids suggesting flags when not applicable
+    manager.commandSuggestionProcessor(
+        new FilteringCommandSuggestionProcessor<>(
+            FilteringCommandSuggestionProcessor.Filter.Simple.contextFree(
+                (s, i) -> i.isEmpty() || s.startsWith("-") == s.startsWith(i))));
```

Comment:
Starts with should be case-insenitive so that `--FOR` can tab-in `--force` the same way `/JO` can tab-in to `/join`